cmd/minikube/cmd/start.go

```diff
@@ -1274,26 +1274,41 @@ func validateRegistryMirror() {
 
 // This function validates if the --image-repository
 // args match the format of registry.cn-hangzhou.aliyuncs.com/google_containers
-func validateImageRepository(imagRepo string) (vaildImageRepo string) {
+// also "<hostname>[:<port>]"
+func validateImageRepository(imageRepo string) (validImageRepo string) {
 
-	if strings.ToLower(imagRepo) == "auto" {
-		vaildImageRepo = "auto"
+	if strings.ToLower(imageRepo) == "auto" {
+		validImageRepo = "auto"
 	}
-	URL, err := url.Parse(imagRepo)
+	URL, err := url.Parse(imageRepo)
+
 	if err != nil {
 		klog.Errorln("Error Parsing URL: ", err)
 	}
-	// tips when imagRepo ended with a trailing /.
-	if strings.HasSuffix(imagRepo, "/") {
+
+	var hasPorts = false
+	var imageRepoPort string
+
+	if URL.Port() != "" && strings.Contains(imageRepo, ":"+URL.Port()) {
+		hasPorts = true
+		imageRepoPort = ":" + URL.Port()
+	}
+	// tips when imageRepo ended with a trailing /.
+	if strings.HasSuffix(imageRepo, "/") {
 		out.Infof("The --image-repository flag your provided ended with a trailing / that could cause conflict in kuberentes, removed automatically")
 	}
-	// tips when imageRepo started with scheme.
+	// tips when imageRepo started with scheme such as http(s).
 	if URL.Scheme != "" {
-		out.Infof("The --image-repository flag your provided contains Scheme: {{.scheme}}, it will be as a domian, removed automatically", out.V{"scheme": URL.Scheme})
+		out.Infof("The --image-repository flag your provided contains Scheme: {{.scheme}}, which will be removed automatically", out.V{"scheme": URL.Scheme})
+	}
+
+	if hasPorts {
+		validImageRepo = URL.Hostname() + imageRepoPort + strings.TrimSuffix(URL.Path, "/")
+	} else {
+		validImageRepo = URL.Hostname() + strings.TrimSuffix(URL.Path, "/")
```

Comment:
This works perfectly fine, but could be streamlined a bit.

The `hasPorts` variable could be removed altogether, if the `if` on 1292 is false, `imageRepoPort` will be empty, so you can always do `validImageRepo = URL.Hostname() + imageRepoPort + strings.TrimSuffix(URL.Path, "/")` and will get the same result as now.